src/main.ts

```diff
@@ -944,32 +944,38 @@ function generateDecode(ctx: Context, fullName: string, messageDesc: DescriptorP
 
     // and then use the snippet to handle repeated fields if necessary
     if (isRepeated(field)) {
-      const maybeNonNullAssertion = ctx.options.useOptionals === "all" ? "!" : "";
+      const initialValue = isMapType(ctx, messageDesc, field) ? (ctx.options.useMapType ? "new Map()" : "{}") : "[]";
+      const maybeInitalize = `message.${fieldName} = message.${fieldName} ?? ${initialValue}`;
```

Comment:
Just noticed that it's always initializing now, but this code can be omitted if `ctx.options.useOptionals !== "all"`

Reply:
Fixed and force-pushed.